src/Agent.Sdk/Knob/AgentKnobs.cs

```diff
@@ -374,6 +374,18 @@ public class AgentKnobs
             new RuntimeKnobSource("DISABLE_BUILD_ARTIFACTS_TO_BLOB"),
             new EnvironmentKnobSource("DISABLE_BUILD_ARTIFACTS_TO_BLOB"),
             new BuiltInDefaultKnobSource("false"));
+        public static readonly Knob SendBuildArtifactsToBlobstoreDomain = new Knob(
```

Comment:
It would be nice to have a Knob to use the previous code path

Reply:
This knob serves double duty -> by setting it to the default domain id (always 0), it overrides everything to use the default domain.  It can also be set to override using the default domain so we can push individual pipelines to alternate domains without having an org-wide setting.

Reply:
That was more of a general comment -- there are a lot of changed code paths.  I know it's designed and tested to fallback gracefully, without a knob to use the old code paths, if there is a bug in the changes, there will be no way to use the old code path other than to rollback the entire agent.  Using a knob to provide the old code path would provide a faster mitigation

This isn't a blocker due to the urgency of the changes + testing, but it would be nice to have